tools/bake.py

```diff
@@ -672,9 +904,52 @@ def first_bsdf(objs):
                     bpy.ops.uv.pack_islands(rotate=True, margin=margin)
                     pass
                 Common.switch('OBJECT')
+            
+            if optimize_solid_materials: 
+                #unhide geometry from step before pack islands that fixed solid material uvs, then scale uv's to be short enough to avoid color squares at top right. - @989onan
+                for child in collection.all_objects:
+                    if child.type == "MESH":           
+                        for layer in cats_uv_layers:
+                            idx = child.data.uv_layers.active_index
+                            child.data.uv_layers[layer].active = True
+                            Common.switch('EDIT')
+                            
+                            bpy.ops.mesh.select_all(action='SELECT')
+                            bpy.ops.uv.select_all(action='SELECT')
+                            
+                            #https://blender.stackexchange.com/a/75095
+                            #Scale a 2D vector v, considering a scale s and a pivot point p
+                            def Scale2D( v, s, p ):
+                                return ( p[0] + s[0]*(v[0] - p[0]), p[1] + s[1]*(v[1] - p[1]) )     
+                            
+                            last_index = len(solidmaterialnames)
+                            
+                            maxium_y_position_materials = ((((int(((last_index*2)+1)/(resolution/4))*-8)+((resolution/2)-4)))/resolution)*2
```

Comment:
Also kinda unreadable

Reply:
It's half the equation from the thing I said on another comment but for the Y cordinate. Here's a quote from me on that one:
"It's an equation to wrap, and I have no idea how to write it better... It's supposed to wrap around the image for the X and Y cords. Do you think there's a better way to write this?"